yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,251 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UnsupportedError,
+    int_or_none,
+    parse_qs,
+    traverse_obj,
+    urljoin,
+)
+
+
+_FIELDS = '''
+    _id
+    clipImageThumb
+    clipLink
+    clipTitle
+    createdDate
+    shareId
+    user { _id }
+    username
+    views'''
+
+_EXTRA_FIELDS = '''
+    clipLength
+    clipSizeBytes'''
+
+_QUERIES = {
+    'clip': '''query ($id: String!) {
+        video: getClip(clipIdentifier: $id) {
+            %s %s
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montage': '''query ($id: String!) {
+        video: getMontage(clipIdentifier: $id) {
+            %s
+        }
+    }''' % _FIELDS,
+    'Clips': '''query ($page: Int!, $user: String!, $game: Int) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: false, game: $game) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'Montages': '''query ($page: Int!, $user: String!) {
+        videos: montages(search: createdDate, page: $page, user: $user) {
+            data { %s }
+        }
+    }''' % _FIELDS,
+    'Mobile Clips': '''query ($page: Int!, $user: String!) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: true) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+}
+
+
+class AllstarBaseIE(InfoExtractor):
+    @staticmethod
+    def _parse_video_data(video_data):
+        def _media_url_or_none(path):
+            return urljoin('https://media.allstar.gg/', path)
+
+        def _profile_url_or_none(path):
+            return urljoin('https://allstar.gg/u/', path)
+
+        return traverse_obj(video_data, {
+            'id': ('_id', {str}),
+            'display_id': ('shareId', {str}),
+            'title': ('clipTitle', {str}),
+            'url': ('clipLink', {_media_url_or_none}),
+            'thumbnail': ('clipImageThumb', {_media_url_or_none}),
+            'duration': ('clipLength', {int_or_none}),
+            'filesize': ('clipSizeBytes', {int_or_none}),
+            'timestamp': ('createdDate', {int_or_none}),
+            'uploader': ('username', {str}),
+            'uploader_id': ('user', '_id', {str}),
+            'uploader_url': ('user', '_id', {_profile_url_or_none}),
+            'view_count': ('views', {int_or_none}),
+        })
+
+    def _send_query(self, query, variables={}, path=(), video_id=None, note=None):
+        response = self._download_json(
+            'https://a1.allstar.gg/graphql', video_id, note=note,
+            headers={'content-type': 'application/json'},
+            data=json.dumps({'variables': variables, 'query': query}).encode())
+
+        errors = response.get('errors')
+        if errors:
+            raise ExtractorError(errors, expected=True)
```

Comment:
If I remember correctly I used the `expected` argument just to silence the report issue message.
errors examples:
in case the montage `id` not found the api returns
`[{'message': 'ID not found', 'locations': [{'line': 2, 'column': 9}], 'path': ['video'], 'extensions': {'code': 'BAD_USER_INPUT'}}]`
and in case of clip `id` not found it returns a different error
`[{'message': "Cannot read properties of null (reading 'comments')", 'locations': [{'line': 2, 'column': 9}], 'path': ['video'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}}]`

Reply:
I don't think we should silence it. If we're getting errors at this point, then something is going wrong with the extractor, yeah? IMO let's do this
```suggestion
        errors = traverse_obj(response, ('errors', ..., 'message', {str}))
        if errors:
            raise ExtractorError('; '.join(errors))
```

Reply:
ok